Fix(Grid): Improve messages around non client-side row models

sdwvit opened the pull request:
# Summary
This PR improves the handling and validation of the rowModelType during grid initialization. 

| Before | After |
|-|-|
| ![Screenshot 2025-05-06 at 16 37 01](https://github.com/user-attachments/assets/f83c6052-665f-44e4-9e84-33b8238d36eb) | ![Screenshot 2025-05-08 at 16 11 45](https://github.com/user-attachments/assets/20a8c85a-00bb-4fd1-b42d-99d40a10542e) ![Screenshot 2025-05-08 at 14 53 05](https://github.com/user-attachments/assets/b6f56689-3eb2-4578-8541-b65972f7861a) |

Fixes https://ag-grid.atlassian.net/browse/AG-14797

## Changes
Error Logging: Added detailed error logging for mismatched or missing modules based on the rowModelType.

As a byproduct of my first change, I am also gitignoring .idea folder. Otherwise all my pr diffs would be huge because it's not intended to be committed :) 

## Testing
- added behavioral tests
- repro steps are to be added to the jira ticket for qa

## Notes
No breaking changes introduced. Existing functionality remains intact.

sdwvit committed:
fix: improve row model type handling and validation in grid initialization

- Added `getDefaultRowModelType` utility to centralize default row model type logic.
- Introduced error message for mismatched or missing modules based on `rowModelType`.

sdwvit committed:
Merge branch 'latest' into fix/sdwvit/grid/server-side-module-err/AG-14797

# Conflicts:
#	packages/ag-grid-community/src/validation/errorMessages/errorText.ts

sdwvit committed:
merge stable

sdwvit committed:
lint

sdwvit committed:
Merge branch 'latest' into fix/sdwvit/grid/server-side-module-err/AG-14797

sdwvit committed:
Update as per https://ag-grid.atlassian.net/browse/AG-14797?focusedCommentId=82617

sdwvit committed:
Update as per https://ag-grid.atlassian.net/browse/AG-14797?focusedCommentId=82617

sdwvit committed:
better variable name

sdwvit committed:
add tests for new error logic

sdwvit committed:
add tests for new error logic

StephenCooper commented on packages/ag-grid-community/src/validation/errorMessages/errorText.ts:
```diff
@@ -672,6 +672,18 @@ export const AG_GRID_ERRORS = {
+        rowModelType: RowModelType | undefined;
+        fallbackRowModelType: RowModelType;
+    }) =>
+        `Module ${moduleName} expects rowModelType '${correctRowModelType}', got ${rowModelType || `nothing (defaults to '${fallbackRowModelType}').`}` as const,
```
Can we share this message to avoid duplication. 

Also I would format it like `expects rowModelType=${correctRowModelType}` so it looks like config. 

Instead of nothing I would go with undefined.

sdwvit committed:
improve test coverage

StephenCooper commented on testing/behavioural/src/grid-api-object/mismatched-options.test.tsx:
```diff
@@ -0,0 +1,72 @@
+    });
+
+    test('Wrong options provided', () => {
+        createMyGrid({ rowModelType: 'infinite' }, { modules: [ServerSideRowModelModule] });
```
Here, the setting of rowModelType should be taken as the intended row model and take priority over the module that is provided.

sdwvit committed:
- share the error message to avoid duplication.
- format it like expects rowModelType=${correctRowModelType} so it looks like config.
- Instead of falsy value generating 'nothing' string use json.stringify().
- update tests to reflect that
- the setting of rowModelType is now taken as the intended row model and take priority over the module that is provided.

sdwvit committed:
- update idea configs

sdwvit commented:
seems like not many ppl using intellij, I have updated configs for latest version

sdwvit committed:
update test

sdwvit committed:
help bundle size

sdwvit committed:
lint

sdwvit committed:
Merge branch 'latest' into fix/sdwvit/grid/server-side-module-err/AG-14797

StephenCooper committed:
Review changes

StephenCooper committed:
Merge branch 'fix/sdwvit/grid/server-side-module-err/AG-14797' of https://github.com/ag-grid/ag-grid into fix/sdwvit/grid/server-side-module-err/AG-14797

sdwvit committed:
add another test case

sdwvit committed:
update tests

sdwvit committed:
Merge branch 'latest' into fix/sdwvit/grid/server-side-module-err/AG-14797

sdwvit committed:
add link to the row models docs on error page

StephenCooper commented on packages/ag-grid-community/src/validation/errorMessages/errorText.ts:
```diff
@@ -119,6 +129,25 @@ const clipboardApiError = (method: string) =>
     'Either fix why Clipboard API is blocked, OR stop this message from appearing by setting grid ' +
     'property suppressClipboardApi=true (which will default the grid to using the workaround rather than the API.';
 
+type ModuleExpectsErrorParams = {
```
This can be removed. Likely a merge issue.

sdwvit committed:
revert grid-api-object.test.ts

sdwvit committed:
remove dead code

sdwvit committed:
kill .idea
